### PR TITLE
Enhanced Mobile Screen Responsiveness

### PR DESCRIPTION
**Pull request by Blazebrain**
-- Replaced listView with DataTable
-- Enhanced responsiveness for mobile screens

### CODE REVIEW

**Comment by escamoteur** [comment]
Uuuahaha, I switched from DataTable to ListView not long ago 😲

The reason to do that is because with more Entries a datatable gets very slow.

I will checkout your changes on the reponsiveness too.
best to hit me on Slack before you make bigger changes and discuss them with me :-)

File: lib/main.dart [hunk 1]
```diff
@@ -124,53 +129,57 @@ class _Header extends StatelessWidget with GetItMixin {
...
-                        'assets/images/devshelpdevs-logo.svg',
-                        height: 100,
-                      ),
-                    ),
-                    Padding(
-                      padding: const EdgeInsets.only(top: 0),
-                      child: TextButton(
-                        onPressed: () async {
-                          await launch('https://paypal.me/pools/c/8xPwkVP3th');
+                child: SingleChildScrollView(
```
Comment (escamoteur) [inline lib/main.dart:132]: Why a scrollview in the Header?

File: lib/presentation/usage.dart [hunk 2]
```diff
@@ -1,140 +1,97 @@
...
-          style: tableHeaderStyle,
-          child: Row(
-            children: [
-              if (hasUsageDates) Expanded(child: Text('Date')),
-              Expanded(child: Text('Amount')),
-              Expanded(child: Text('Usage')),
-              Spacer(),
-              Expanded(child: Text('Receiver')),
-              Spacer(),
+    final columnSpacing = LayoutValue.fromBreakpoint(
```
Comment (escamoteur) [inline lib/presentation/usage.dart:24]: I like that!

File: lib/presentation/usage.dart [hunk 3]
```diff
@@ -1,140 +1,97 @@
...
+      xs: 80.0,
+      md: 64.0,
+      lg: 56.0,
+      xl: 48.0,
+    );
+    return SingleChildScrollView(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          DataTable(
```
Comment (escamoteur) [inline lib/presentation/usage.dart:40]: As already stated, we should stay with a ListView here.
What you think if we switch for mobiles from a table layout to a card like layout that allows more than one line per entry?
I added a second optional picture to the Usage class in the meantime which we also have to place somewhere if provided.

File: lib/main.dart (hunk 1 above)
Comment (Blazebrain) [reply lib/main.dart:132]: Good day sir,

The header line overflows the width of the screen and for now, I just decided to give it a scroll view to make it scrollable.

However, in my next commit, I plan to make the header adapt in size either through the use of layout breakpoints or specifying the width per screen width type

Comment (escamoteur) [reply lib/main.dart:132]: Ah. I missed that it's a horizontal one.
Probably we have to use a totally different header widget for mobile

Comment (Blazebrain) [reply lib/main.dart:132]: That's a more preferable option sir

**Comment by Blazebrain** [comment]
> Uuuahaha, I switched from DataTable to ListView not long ago astonished
> 
> The reason to do that is because with more Entries a datatable gets very slow.
> 
> I will checkout your changes on the reponsiveness too.
> best to hit me on Slack before you make bigger changes and discuss them with me :-)

Alright sir, definitely

File: lib/presentation/usage.dart (hunk 3 above)
Comment (Blazebrain) [reply lib/presentation/usage.dart:40]: Nice, it would enable the full-text show completely on the screen which is a better alternative

**Comment by escamoteur** [comment]
Hi, I made already some responsiveness changes yesterday so I close the PR here

**Pull request closed without merging**